rest/server_context.go

```diff
@@ -292,8 +294,18 @@ func (sc *ServerContext) GetInactiveDatabase(ctx context.Context, name string) (
 			}
 		}
 	}
+	// handle the correct error message being returned for a corrupt database config
+	var httpErr *base.HTTPError
+	sc.lock.RLock()
+	_, ok := sc.corruptDbContext[name]
+	sc.lock.RUnlock()
+	if !dbConfigFound && ok {
+		httpErr = base.HTTPErrorf(http.StatusNotFound, "Database config corrupt. Database config bucket name not equal to bucket it is persisted in. Must update database config immediately")
```

Comment:
I think that we should add the names of the buckets and database names here, if this code has to stay, which I'm not sure it does.

Reply:
So this code needs to stay to return the correct error message for a corrupt db config, will add the extra info to error message though